queue/server.go

```diff
@@ -0,0 +1,123 @@
+package queue
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	vulcanreport "github.com/adevinta/vulcan-report"
+)
+
+// CheckState defines the fields of a check relevant during its execution.
+type CheckState struct {
+	Status   string              `json:"status,omitempty"`
+	Progress float32             `json:"progress,omitempty"`
+	Report   vulcanreport.Report `json:"report,omitempty"`
+	Error    string              `json:"error,omitempty"`
+}
+
+// Check defines the relevant informartion for the runtime that executes and
+// monitors checks.
+type Check struct {
+	ID    string
+	State CheckState
+}
+
+type Queue interface {
+	Dequeue() (*Check, error)
+	Enqueue(Check) error
+}
+
+type SimpleMQClientServer struct {
+	q    *SimpleMQ
+	Addr string
+}
+
+func NewSimpleMQClientServer() (SimpleMQClientServer, error) {
+	addr, err := freeLocalAddr()
+	if err != nil {
+		return SimpleMQClientServer{}, err
+	}
+
+	q := New(addr, "check")
+	s := SimpleMQClientServer{
+		q:    q,
+		Addr: fmt.Sprintf("http://%s", addr),
+	}
+	return s, nil
+}
+
+func (s SimpleMQClientServer) Start() error {
+	return s.q.ListenAndServe()
+
+}
+
+func (s SimpleMQClientServer) WaitStart(timeout time.Duration) error {
+	d := time.NewTimer(timeout)
+	for {
+		select {
+		case <-d.C:
+			return errors.New("timeout waiting for the queue to start")
+		default:
+			c, err := net.Dial("tcp", strings.Replace(s.Addr, "http://", "", -1))
```

Comment:
Yep good sugestiion.

Reply:
Well no!! I mean if for whatever reason the Dial fails, the the timeout will fire and we will return and error. It is true that the error won't be very descriptive, because it will be a timeout error instead of a "dial error". I think that taking into account this functionality is, let say, only to run locally the check while developing, and we will probably are going to change it in near future, it is acceptable. What do you think?